services/private_ip_manager/src/main/java/com/futurewei/alcor/privateipmanager/repo/IpAddrRangeRepo.java

```diff
@@ -0,0 +1,259 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+
+package com.futurewei.alcor.privateipmanager.repo;
+
+import com.futurewei.alcor.common.db.CacheFactory;
+import com.futurewei.alcor.common.db.ICache;
+import com.futurewei.alcor.common.db.Transaction;
+import com.futurewei.alcor.common.repo.ICacheRepository;
+import com.futurewei.alcor.privateipmanager.entity.IpAddrAlloc;
+import com.futurewei.alcor.privateipmanager.entity.IpAddrRange;
+import com.futurewei.alcor.privateipmanager.entity.IpAddrRangeRequest;
+import com.futurewei.alcor.privateipmanager.exception.InternalDbOperationException;
+import com.futurewei.alcor.privateipmanager.exception.IpAddrRangeNotFoundException;
+import com.futurewei.alcor.privateipmanager.exception.IpAddrRangeExistException;
+import com.futurewei.alcor.privateipmanager.exception.SubnetNotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.stereotype.Repository;
+import com.futurewei.alcor.common.db.CacheException;
+import javax.annotation.PostConstruct;
+import java.util.*;
+
+@ComponentScan(value="com.futurewei.alcor.common.db")
+@Repository
+public class IpAddrRangeRepo implements ICacheRepository<IpAddrRange> {
+    private static final Logger LOG = LoggerFactory.getLogger(IpAddrRangeRepo.class);
+    private ICache<String, IpAddrRange> ipAddrRangeCache;
```

Comment:
The key here is the subnet Id. How about multiple ip ranges per subnet?

Reply:
I think an ip range is actually a subnet , maybe i should called it rangeId instead of subnetId?

Reply:
Hmm, a subnet could have at least one Ipv4 range and one Ipv6 range.

Reply:
Use RangeId as the key sounds better. We could add an ID field in the IpAddrRange class.